dragonfly/session/text.go

```diff
@@ -134,3 +134,13 @@ func (s *Session) SendSubtitle(text string) {
 func (s *Session) SendActionBarMessage(text string) {
 	s.writePacket(&packet.SetTitle{ActionType: packet.TitleActionSetActionBar, Text: text})
 }
+
+// SetJoinMessage changes the message that appears when the player joins the server
+func (s *Session) SetJoinMessage(message string) {
+	s.joinMessage = message
```

Comment:
I think these properties are vulnerable to race conditions. Consider changing them to an `atomic.String`